src/core/logger.js

```diff
@@ -0,0 +1,16 @@
+// evaluation of DEV needs to be done inside the functions
+// otherwise console.log cannot be changed to a spy function anymore.
+
+export function info(msg) {
```

Comment:
OK maybe that comment is misleading. What is meant is that once the module is imported, NODE_ENV is evaluted and the calls are either routed to console.log or not. Once fixed to console.log, it cannot be altered by a test anymore (in the sense that it is NOT called whatsoever) since that would require "running" the module again, evaluating NODE_ENV again after the test changed it.

My intention is to have tests that actually check that console.log is called when NODE_ENV==="test"|"development", and NOT called otherwise.

Reply:
I think you can accomplish this by some combination of `jest.resetModules()` and a `require()` statement to import the module (don't use the ES5 module import syntax).

Reply:
```
beforeEach(() => {
  jest.resetModules();   // this may or may not be actually necessary.
});

test1(() => {
  process.env.NODE_ENV = 'development';
  const logger = require('./logger');
  ...
});

test1(() => {
  process.env.NODE_ENV = 'production';
  const logger = require('./logger');
  ...
});
```

Reply:
Using two separate test files is also an option if this doesn't work.